lib/v1alpha/network.go

```diff
@@ -117,38 +127,89 @@ func GetInterfaces() (*types.InterfaceResponse, error) {
 			glog.Infof("  k:%v v:%v", k, v)
 			valueParts := strings.Split(string(v), ",")
 			for _, id := range valueParts {
-				glog.Infof("     Adding ID:%v", id)
-				pciAddressSlice = append(pciAddressSlice, id)
+				found := false
+				// DeviceInfo in the NetworkStatus annotation also has PCI Address.
+				// So skip if PCI Address already found.
+				for _, ifaceData := range response.Interface {
+					if ifaceData.NetworkStatus.DeviceInfo != nil &&
+						ifaceData.NetworkStatus.DeviceInfo.Pci != nil {
+						if strings.EqualFold(ifaceData.NetworkStatus.DeviceInfo.Pci.PciAddress, id) {
+							// PCI Address in ENV matched that in DeviceInfo. Mark as SR-IOV.
+							ifaceData.DeviceType = types.INTERFACE_TYPE_SRIOV
+							found = true
+							break
+						}
+					}
+				}
+				if found {
+					glog.Infof("     Skip Adding ID:%v", id)
+				} else {
+					glog.Infof("     Adding ID:%v", id)
+					pciAddressSlice = append(pciAddressSlice, id)
+				}
 			}
 		}
 	}
 
+	// PCI Addresses were
```

Comment:
This annotate seems not finished?

Reply:
Fixed on next commit. Added a loop to count how many "unknowns" there were.